### PR TITLE
Azure content filter error

### PR DESCRIPTION
**Pull request by R0bk**
Hi all,

When trying to use GPT3.5 and GPT4 for a project I've been running into a common non descriptive error 
```
"...langchain/chat_models/openai.py", line 75, in _convert_dict_to_message
    return AIMessage(content=_dict["content"])
KeyError: 'content'
```

Just doing a little bit of debug shows this is due to Azure's content filtering policies seen [here](https://learn.microsoft.com/en-us/azure/cognitive-services/openai/concepts/content-filter).

I've just added an error that highlights when these have been hit that it is due to the policy, instead of failing without a descriptive error. I couldn't find any other error types that have been setup in the repo for content filters in specific so I used a value error but happy to add a new type if that would be cleaner.

### CODE REVIEW

File: langchain/chat_models/azure_openai.py [hunk 1]
```diff
@@ -110,3 +110,15 @@ def _default_params(self) -> Dict[str, Any]:
...
         }
+
+    def _create_chat_result(self, response: Mapping[str, Any]) -> ChatResult:
+        generations = []
+        for res in response["choices"]:
+            if "finish_reason" in res and res["finish_reason"] == "content_filter":
+                raise ValueError("Azure has not provided the response due to a content"
+                                 " filter being triggered")
+            message = _convert_dict_to_message(res["message"])
+            gen = ChatGeneration(message=message)
```
Comment (pranjaldoshi96) [inline langchain/chat_models/azure_openai.py:121]: Why we don't need to import `ChatGeneration` they need to import it for base class ChatOpenAI as well https://github.com/hwchase17/langchain/blob/master/langchain/chat_models/openai.py#L25

File: langchain/chat_models/azure_openai.py [hunk 2]
```diff
@@ -110,3 +110,15 @@ def _default_params(self) -> Dict[str, Any]:
             **super()._default_params,
             "engine": self.deployment_name,
         }
+
+    def _create_chat_result(self, response: Mapping[str, Any]) -> ChatResult:
+        generations = []
+        for res in response["choices"]:
+            if "finish_reason" in res and res["finish_reason"] == "content_filter":
```
Comment (pranjaldoshi96) [inline langchain/chat_models/azure_openai.py:117]: You can use res.get()
```
if res.get("finish_reason", None) == "content_filter":
```

File: langchain/chat_models/azure_openai.py [hunk 3]
```diff
@@ -110,3 +110,15 @@ def _default_params(self) -> Dict[str, Any]:
...
+                raise ValueError("Azure has not provided the response due to a content"
+                                 " filter being triggered")
+            message = _convert_dict_to_message(res["message"])
+            gen = ChatGeneration(message=message)
+            generations.append(gen)
+        llm_output = {"token_usage": response["usage"], "model_name": self.model_name}
+        return ChatResult(generations=generations, llm_output=llm_output)
```
Comment (pranjaldoshi96) [inline langchain/chat_models/azure_openai.py:124]: For the same reason as above you need to import ChatResult

File: langchain/chat_models/azure_openai.py (hunk 1 above)
Comment (R0bk) [reply langchain/chat_models/azure_openai.py:125]: Imported 👍

File: langchain/chat_models/azure_openai.py (hunk 3 above)
Comment (R0bk) [reply langchain/chat_models/azure_openai.py:128]: Imported 👍

**Review by pranjaldoshi96 — APPROVED** [review]

**Pull request merged**